### DISCUSSION TITLE
Binary Caching and Github Actions

### DISCUSSION THREAD

**Discussion by justusranvier** [microsoft/vcpkg#31272] [Q&A]
I've found the documentation for binary caching here to be less than helpful: https://learn.microsoft.com/en-us/vcpkg/users/binarycaching

According to that article if I want compiled libraries to be available for CI jobs and also for developers building on their own machines then I need NuGet packages stored somewhere, probably on GitHub Packages.

The article is fairly clear about how to configure a build to use NuGet packages but says almost nothing about how to create them.

1. Is GitHub Packages something I have to sign up for? Where to I find more information about how to enable it? How many packages can be stored? If I want to cache a large library like qtdeclarative and its dependencies for several triplets is it going to run out of capacity?

2. How do tell vcpkg to create and upload the NuGet packages? I have several projects that are all built with cmake and have a manifest (vcpkg.json) Is there some cmake option I can pass that will tell vcpkg to generate the necessary packages and upload them?

**Comment by funnan-dev** [comment]
@justusranvier Hey, were you able to figure this out? I have set up a number of pipelines for my company using NuGet and can hopefully answer your question.

What is boils down to, is you need to configure your NuGet locally to point to a remote NuGet source. This step has nothing to do with vcpkg so you may need to track down documentation on this itself.

Once you've set this up, with a name, let's say, "NuVcpkg". Then you should be able to use vcpkg's binary caching flag to push/pull as needed.

`--binarysource="nuget,NuVcpkg,readwrite"`

**Reply by justusranvier** [reply]
The underlying issue is that I don't know what NuGet is and I can't find any information about it that's not specific to .NET framework, which I'm not using.

Is NuGet exclusive to .NET, or can it be used for regular c++ projects?
